### ISSUE TITLE
[BUG] PagedIterable<BlobItem> stream().paralllel() is behaving as sequential

### ISSUE THREAD

**Issue by varenyavv** [Azure/azure-sdk-for-java#40768]
**Describe the bug**
PagedIterable\<BlobItem\> stream().paralllel() is behaving as sequential.

***Exception or Stack Trace***
Log showing a single forkjoinpool worker is performing the task even though it has 20 workers.
```
2024-06-21 17:59:21,275 INFO  [ForkJoinPool.commonPool-worker-1] c.o.g.f.v.s.o.IndividualRequestXmlAggregator gpfh-vendor-outbound-job batchId: '2000' sourceSystem: 'GPS': Downloading 
blob 789/generated/2023-12-03/1232939632.xml
2024-06-21 17:59:22,208 INFO  [ForkJoinPool.commonPool-worker-1] c.o.g.f.v.s.o.IndividualRequestXmlAggregator gpfh-vendor-outbound-job batchId: '2000' sourceSystem: 'GPS': Downloading 
blob 789/generated/2023-12-03/1232939814.xml
2024-06-21 17:59:23,143 INFO  [ForkJoinPool.commonPool-worker-1] c.o.g.f.v.s.o.IndividualRequestXmlAggregator gpfh-vendor-outbound-job batchId: '2000' sourceSystem: 'GPS': Downloading 
blob 789/generated/2023-12-03/1232939895.xml
2024-06-21 17:59:24,065 INFO  [ForkJoinPool.commonPool-worker-1] c.o.g.f.v.s.o.IndividualRequestXmlAggregator gpfh-vendor-outbound-job batchId: '2000' sourceSystem: 'GPS': Downloading 
blob 789/generated/2023-12-03/1232939934.xml
```

**To Reproduce**
Use the below code snippet to list the blobs. Logs will display that the task is occurring serially which is causing slowness if the blob counts are in millions.

***Code Snippet***
```
BlobServiceClient blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionString).buildClient();
BlobContainerClient blobContainerClient = blobServiceClient.getBlobContainerClient(blobContainerName);
String blobPrefix = "789/generated/";
ListBlobsOptions options = new ListBlobsOptions().setPrefix(blobPrefix).setMaxResultsPerPage(5000);
PagedIterable<BlobItem> blobItems = blobContainerClient.listBlobs(options, null, null);
blobItems.stream()
        .parallel()
        .filter(this::isAnXml)
        .forEach(
            blobItem -> {
              LOGGER.info("Downloading blob {}", blobItem.getName());
              //more business logic
            });
```

**Expected behavior**
Execution should happen in parallel by multiple worker threads rather than by the single thread in sequential manner.

**Setup (please complete the following information):**
OS: Ubuntu 22.04.4 LTS
IDE: Intellij
Library/Libraries: com.azure:azure-storage-blob:12.25.0
Java version: Openjdk version 17.0.10
App Server/Environment: Tomcat embedded in Springboot
Frameworks: Springboot v3.2.3

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by alexfandos** [comment]
Do you happen to have any news on this? I just stumbled across the same issue.

**Comment by browndav-msft** [comment]
@varenyavv , @alexfandos ,
I'm getting the same thing you both are getting.

<img width="340" height="265" alt="Image" src="https://github.com/user-attachments/assets/18405f3f-ac9b-4bb5-bef4-d11d8cddd52b" />

 I'm going to talk to someone on the team about this and will get back to you.

**Comment by browndav-msft** [comment]
@varenyavv @alexfandos 

I did some more tinkering and see that if I run this in my IDE (IntelliJ) then everything runs on a single thread; however, if I run it from the command line, then I get multiple threads.

<img width="645" height="230" alt="Image" src="https://github.com/user-attachments/assets/2a70707d-56f8-4208-b33c-ab8edec0ea3a" />

**Comment by browndav-msft** [comment]
@varenyavv @alexfandos

I'm not sure whether you are still experiencing the issue, but I'm going to close this ticket. Please let us know if you are still experiencing the issue by opening another ticket.
